File: DSA/PalakRastogi_590018052/Day2Question3.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;


bool arrEqual(vector<int> &arr1, vector<int> &arr2)
{
    sort(arr1.begin(),arr1.end());
    sort(arr2.begin(),arr2.end());

    if(arr1.size()!=arr2.size())
    return false;

    for(int i=0; i<arr1.size(); i++)
    {
        if(arr1[i]!=arr2[i])
        return false;
    }

    return true;
}


int main()
{
    vector<int> arr1={1,2,3,4};

    //vector<int> arr2={3,4,2,1};

    vector<int> arr2={1,2,3};

    if(arrEqual(arr1,arr2))
    {
        cout << "true";

        return 0;
    }

    cout << "false";
    return 0;

}

File: DSA/VaniJain_590012020/Day2Question3.java
import java.util.Arrays;
public class ArraysAreEqualSimple 
{
  public static boolean areEqual(int[] arr1, int[] arr2) 
  {
        if (arr1.length!=arr2.length) 
        {
            return false;
        }
        Arrays.sort(arr1);
        Arrays.sort(arr2);
        return Arrays.equals(arr1, arr2);
    }
    public static void main(String[] args) 
    {
        int[] arr1 = {1, 2, 3, 2};
        int[] arr2 = {2, 3, 1, 2};
        int[] arr3 = {1, 2, 3};
        int[] arr4 = {3, 2, 2};
        System.out.println("Arrays are equal: " + areEqual(arr1, arr2));
        System.out.println("Arrays are equal: " + areEqual(arr3, arr4)); 
    }
}
